### PR TITLE
Create workspace devfile

### PR DESCRIPTION
**Pull request by michael-valdron**
### What does this PR do?:
_Summarize the changes. Are any stacks or samples added or updated?_

Adds a devfile and resource files to build and deploy the community devfile registry.

Additionally adds ignore for odo project dotfile.

### Which issue(s) this PR fixes:
_Link to github issue(s)_

fixes devfile/api#1200

### PR acceptance criteria:

- [x] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [x] Test automation
_Does this repository's tests pass with your changes?_
- [x] Documentation
_Does any documentation need to be updated with your changes?_
- [x] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,34 @@ If you are a stack owner and need to request an urgent refresh of <https://regis
...
+- Docker 17.06 or later
+- Git
+- Kubernetes or Red Hat OpenShift
+- odo v3.15.0+
+
+### Deploying
+
+The following can be used to build and deploy a devfile registry locally:
+
+odo V3: `odo deploy`
```
Comment (thepetk) [inline README.md:49]: I've tried to deploy locally using `minikube` & `odo v3` but I have the error:

```
Error: writing blob: initiating layer upload to /v2/devfile/
devfile-index/blobs/uploads/ in quay.io: unauthorized:
access to the requested resource is not authorized
```

File: README.md [hunk 2]
```diff
@@ -31,7 +31,34 @@ If you are a stack owner and need to request an urgent refresh of <https://regis
...
+
+Prevent odo v3 deployment built images from being pushed by running:
+
+```sh
+ODO_PUSH_IMAGES=false odo deploy --var indexPullPolicy=Never
```
Comment (thepetk) [inline README.md:54]: Tried also this command but post deployment I get an `ErrImageNeverPull`

**Review by thepetk — COMMENTED** [review]
Tried to deploy locally but I get some errors. Might be related to my setup.

File: README.md (hunk 1 above)
Comment (michael-valdron) [reply README.md:49]: I'll update this to indicate to specify the `--var indexImageName=<write_access_image> --var indexImageTag=<write_access_imagetag>`, this wasn't update before the ODC support commit I did.

Comment (michael-valdron) [reply README.md]: Should be resolved in https://github.com/devfile/registry/pull/256/commits/3f4316c9976c0081034825e82ddfb97c17e63c27.

File: README.md [hunk 3]
```diff
@@ -31,7 +31,43 @@ If you are a stack owner and need to request an urgent refresh of <https://regis
...
+- odo v3.15.0+
+
+### Deploying
+
+The following can be used to build and deploy a devfile registry locally:
+
+odo V3: 
+
+```sh
+odo deploy --var indexImageName=quay.io/<user>/devfile-index --var indexImageTag=<tag>
```
Comment (thepetk) [inline README.md:52]: Should we also add here the `hostName` variable and suggest a custom host name for local deployments?

**Review by thepetk — COMMENTED** [review]
Locally looks good after the updates and I was able to deploy the registry pretty easy. I left a quick comment for the readme :)

File: README.md (hunk 3 above)
Comment (michael-valdron) [reply README.md:52]: Should be resolved in https://github.com/devfile/registry/pull/256/commits/4d7dc5e5b31c6ca09d15ddee9712f97a70f2ba4e.

**Review by thepetk — APPROVED** [review]
/lgtm

**Pull request merged**